### ISSUE TITLE
Async subqueries returns the wrong results

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#1964]
It looks like an async subquery uses the original Table instead of the TableView to filter queries, which result in the async query returning the wrong result.

Unit test exposing it: https://github.com/realm/realm-java/commit/20293bcfb4585208cf54cfe4d2dcc99f4c9f3f5c

```
// Subquery will search all Foo objects, not just those in `foos`
RealmResults<Foo> foos = realm.where(Foo.class).findAll();
RealmResults<Foo> subQueryFoos = foos.where().equalTo("field", value).findAllAsync();
```
/cc @nhachicha

**Comment by cmelchior** [comment]
Fixed by https://github.com/realm/realm-core/pull/1420. Awaits a new core release.

**Comment by nhachicha** [comment]
fixed in `0.87.1`
